common/animation.go

```diff
@@ -80,41 +94,39 @@ func (ac *AnimationComponent) NextFrame() {
 	}
 }
 
+// AnimationSystem tracks AnimationComponents, advancing their current animation.
+type AnimationSystem struct {
+	entities map[ecs.BasicEntity]animationEntity
+}
+
 type animationEntity struct {
-	*ecs.BasicEntity
 	*AnimationComponent
 	*RenderComponent
 }
 
-type AnimationSystem struct {
-	entities []animationEntity
-}
-
+// Add starts tracking the given entity.
 func (a *AnimationSystem) Add(basic *ecs.BasicEntity, anim *AnimationComponent, render *RenderComponent) {
-	a.entities = append(a.entities, animationEntity{basic, anim, render})
+	if a.entities == nil {
+		a.entities = make(map[ecs.BasicEntity]animationEntity)
+	}
+	a.entities[*basic] = animationEntity{anim, render}
 }
 
+// Remove stops tracking the given entity.
 func (a *AnimationSystem) Remove(basic ecs.BasicEntity) {
-	delete := -1
-	for index, e := range a.entities {
-		if e.BasicEntity.ID() == basic.ID() {
-			delete = index
-			break
-		}
-	}
-	if delete >= 0 {
-		a.entities = append(a.entities[:delete], a.entities[delete+1:]...)
+	if a.entities != nil {
```

Comment:
That's true, but it also means I can't add entities before adding the system to World. It'd be nice to keep that working.